TripCheckins/Add trip/AddTripViewController.swift

```diff
@@ -40,18 +50,26 @@ class AddTripViewController: UIViewController {
         navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel,
                                                            target: self,
                                                            action: #selector(cancelButtonTapped(_:)))
+        view.addSubview(titleTextField)
         view.addSubview(dateFilterCreationView)
     }
     
     override func viewDidLayoutSubviews() {
         super.viewDidLayoutSubviews()
-        dateFilterCreationView.frame = CGRect(x: 0, y: 100, width: view.frame.width, height: 50)
+        titleTextField.frame = CGRect(x: 11, y: 100, width: view.frame.width - 22, height: 50)
+        dateFilterCreationView.frame = CGRect(x: 0, y: 170, width: view.frame.width, height: 50)
     }
 
     // MARK: Actions
     @objc func doneButtonTapped(_ sender: Any) {
         let dateFilter = dateFilterCreationView.currentDateFilter
-        delegate?.addTripControllerDidTriggerAddAction(self, dateFilter: dateFilter)
+        var title = "New trip"
```

Comment:
I think it will be better to define the default title for the trip in the trip initialiser than in the view controller, it will also allow you to get rid of the next if let

Reply:
I don't want to have this string as default trip title, using this placeholder for now until I implement the TODO from above that won't allow creating a trip without title. Plus even if I move the code to the `Trip` file I will have to keep the if because empty text fields actually return empty strings for `text`, not nil :/